aten/src/ATen/native/Activation.cpp

```diff
@@ -43,6 +43,205 @@ Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Genera
   return at::rrelu_with_noise_(self, self.type().tensor(), lower, upper, training, generator);
 }
 
+// -----------------------------------
+// prelu forward
+// -----------------------------------
+template <typename scalar_t>
+void prelu_cpu_kernel(
+  Tensor& result,
+  const Tensor& input,
+  const Tensor& weight,
+  int64_t input_dim0_size,
+  int64_t channel_size,
+  int64_t input_stride0,
+  int64_t input_stride1) {
+
+  int64_t i, j, k;
+  auto result_data = result.data<scalar_t>();
+  auto input_data = input.data<scalar_t>();
+  auto weight_data = weight.data<scalar_t>();
+
+  // multiply values at each channel with weight[channel_index]
+  #pragma omp parallel for private(i,j,k) if (input.numel() > 10000)
+  for (i = 0; i < input_dim0_size; i++) {
+    for (j = 0; j < channel_size; j++) {
+      for (k = 0; k < input_stride1; k++) {
+        int64_t pos = i * input_stride0 + j * input_stride1 + k;
+        result_data[pos] = (input_data[pos] > 0) ? input_data[pos] : weight_data[j] * input_data[pos];
+      }
+    }
+  }
+}
+
+Tensor prelu_cpu(const Tensor& self, const Tensor& weight_) {
+  auto input = self.contiguous();
+  auto weight = weight_.contiguous();
+
+  AT_CHECK(input.is_contiguous());
+  AT_CHECK(weight.is_contiguous());
+
+  int64_t weight_num = weight.numel();
+  Tensor result = at::empty_like(input);
+  auto strides = input.strides();
+
+  // case1: shared weight for all channels
+  if (weight_num == 1) {
```

Comment:
@ezyang I mostly just copy&past from previous implementation, why there will be such a big performance gap?
https://github.com/pytorch/pytorch/blob/3da8d71d7de6f2eb3c9bbe395e38dc2fdf2784b0/aten/src/THNN/generic/PReLU.c#L5-L23

Reply:
this is fixed by parallelizing the kernel